### ISSUE TITLE
representing the z-dimension: elevation, depth, etc.

### ISSUE THREAD

**Issue by mbjones** [ESIPFed/science-on-schema.org#133]
The [POLDER community](https://polder.info) (cc @pbricher @rduerr @ahayes) asked to clarify how to indicate the z-dimension in spatial coverage metadata. This issue is mentioned in the other issues on spatial coverage, including issues #8, #105, #101, and #130. But at this point specifically how to encode elevation and depth may not be entirely clear, especially as some of these are relative to an arbitrary reference point which would also need to be provided (e.g., height above ground for sensor on a tower installation). So POLDER has requested that the guidelines be explicit about these issues.

The `schema:GeoCoordinates` and `schema:GeoShape` already allow a [`schema:elevation`](https://schema.org/elevation) property, but this structure implies WGS84. The value can either include a unit (e.g., `976 m`), or it assumed to be in meters.  From the definition:

> The elevation of a location (WGS 84). Values may be of the form 'NUMBER UNITOFMEASUREMENT' (e.g., '1,000 m', '3,200 ft') while numbers alone should be assumed to be a value in meters. 

Here's an example of use:

```json
{
  "@context": {
    "@vocab": "https://schema.org/"
  },
  "@type": "Dataset",
  "name": "Removal of organic carbon by natural bacterioplankton communities as a function of pCO2 from laboratory experiments between 2012 and 2016",
  ...
  "spatialCoverage": {
    "@type": "Place",
    "geo": {
      "@type": "GeoCoordinates",
      "latitude": 39.3280,
      "longitude": 120.1633,
      "elevation": 1000
    }
  }
}
```

So, questions:

- is `schema:elevation` sufficiently expressive as is?
- do we need to be able to express `depth` as more than negative elevation?
- do we need to be able to express depth and elevation relative to alternative points of reference than WGS84, especially for e.g., depth below mean sea level as an orthometric height? 

@pbricher @rduerr @ahayes please clarify if I missed issues that arose and that affect this discussion. My initial take is that using WGS84-based elevation values is likely sufficient, but you might have alternate use cases that would demonstrate otherwise. Thanks!

**Comment by pbricher** [comment]
I think this summarises the question well, thanks Matt. I've reached out to a colleague involved in GEBCO to see if the bathymetry community have thoughts on this.

**Comment by smrgeoinfo** [comment]
The [SOSO recommendations for Spatial Coverage](https://github.com/ESIPFed/science-on-schema.org/blob/master/guides/Dataset.md#spatial-coverage) suggest using additionalProperty to add spatial reference system information.   EPSG registry has various [vertical datums](https://epsg.io/?q=vertical%20datum%20kind:DATUM).  WGS 84 doesn't define a vertical reference system.

**Comment by njarboe** [comment]
For the MagIC paleomagnetic data model (https://www2.earthref.org/MagIC/data-models/3.0) we have these geography data fields for our sites table that would be nice to be able to describe in schema.org:
3.23 Latitude - lat - Number in Degrees - Site geographic location, Latitude **(included above)**
3.24 Latitude Sigma - lat_sigma - Number in Degrees - Site geographic location averaging uncertainty, Latitude
3.25 Longitude - lon - Number in Degrees - Site geographic location, Longitude **(included above)**
3.26 Longitude Sigma, lon_sigma - Number in DegreesSite geographic location averaging uncertainty, Longitude
3.27 Geographic Precision, geographic_precision - Number in Degrees - Site geographic location, Precision in latitude and longitude
3.28 Stratigraphic Order, order - Integer - Site geographic location, Order in the stratigraphic section
3.29 Stratigraphic Height, height - Number in m - Site geographic location, Stratigraphic height
3.30 Elevation, elevation - Number in m - Site geographic location, Elevation relative to sea level
3.31 Core Depth, core_depth - Number in m - Site geographic location, Core depth below seafloor/lake bottom or ground surface
3.32 Composite Depth, composite_depth - Number in m - Site geographic location, Composite core depth below seafloor/lake bottom or surface

**Comment by mbjones** [comment]
Thanks, @njarboe Can you expand on how you specify the relative depth or elevation reference points in your list above? For example, `depth below seafloor` is a great example of what this issue is trying to address. Thanks.

**Comment by dr-shorthair** [comment]
Vertical datum is usually an ellipsoid. There are many, providing best-fit in different parts of the world - https://epsg.io/?q=ellipsoid%20kind%3AELLIPSOID . Geodesy is complicated. 

 The compromise in accuracy by not specifying this in detail may or may not be important depending on your application. Maybe not for paleomag.

**Comment by njarboe** [comment]
@mbjones In sample based geology, relative position is almost always more important than absolute global position.

Here is a more detailed explanation of these parameters:
In our case height is given in meters above the base of a geologic section
You would might have a series of latitude, longitude, and elevation measurements where one collected samples up a geologic section, but these GPS measurements are not precise enough for doing the detailed stratigraphic studies one is interested in. The height of the beds could differ by centimeters and these heights are measured with rulers in the field for hight precision. This information is critical to understanding the data and needs to be included in the meta data. 

I think units will need to be added to theses fields also.

stratigraphic order is the order of sampling, usually going from oldest in age to youngest. Often the height is not measured in the field as it can be difficult, dangerous, and time consuming. But how the data is ordered stratigraphically can still be recorded and is an important feature of the data set.

core_depth  - absolute position would be the height of the surface (elevation on land and sea floor elevation) minus the core_depth

composite_depth - same as core_depth but this depth is determined by combining the information from many cores drilled in the same location to create a more accurate picture of the geology of the area.

Is this what you meant by specify? We define these terms in our data model at
https://www.earthref.org/MagIC/data-models/3.0
and a .json version can be downloaded there.

**Comment by datadavev** [comment]
Would `variableMeasured` be an appropriate mechanism to describe such observations? Perhaps something along the lines of:

```
{
  "@context": {
    "@vocab": "https://schema.org/"
  },
  "@id": "z_offset_01",
  "@type": "Dataset",
  "description": "Sample taken 10.43m from start of core.",
  "spatialCoverage": {
    "@id": "place_01",
    "@type": "Place",
    "description": "Core start location. Samples are offset from this location.",
    "geo": {
      "@type": "GeoCoordinates",
      "latitude": 39.3280,
      "longitude": 120.1633,
      "elevation": -1000
    }
  },
  "variableMeasured": [
    {
      "@type": "PropertyValue",
      "name": "core_depth",
      "description": "Sample offset from location",
      "unitText": "meters",
      "value": 10.43
    }
  ]
}
```

**Comment by dr-shorthair** [comment]
Ha ha. Yes, that is a solution. As it is for all positional information, including standard lat-long-elevation etc. I know that Mark Sch. and Matt Jones have advocated this in the past (I think it might have been the OBOE approach?) 

In Geospatial data the _tradition_ is to manage location separately from other attributes. And of course there is an ISO standard for 'Linear Referencing' https://www.iso.org/standard/32566.html

**Comment by mbjones** [comment]
Yeah, I can see that too, but I don't think representing it in the data is really the objective here. I think people are looking at how to effectively summarize elevation and depth in metadata for discovery purposes, taking into account that many of these are not measured against an absolute datum, as @njarboe's example shows. If we have an absolute datum, then the problem is much simpler and the existing recs suffice I think. Its when we are talking relative position that we need to decide what to do (e.g., height on tower, height above ground, height below canopy, depth in borehole, etc.). Can we summarize those? Is it useful to do so in schema.org for discovery?

P.S. if the elevation data do across each observation inside a Dataset, then our metadata might be better suited to show the range of elevations covered in the Dataset, rather than an absolute vertical point. That is, we shouldn't assume that all Observations in a Dataset are at a fixed vertical elevation.

**Comment by smrgeoinfo** [comment]
if a dataset is about sampling features that are located  in 3-D
1) single sampling feature or 2) set of sampling features, each with associated observation result. 
we are interested in 'location' as a means for data integration, whether is is a range of locations or a single location.  In some cases a location in an Earth Reference Frame is important, e.g. integrating seismic, gravity, mag and existing borehole data to target drilling an oil well that's going to cost $500M.   I the cases @njarboe mentions, relative location w.r.t to stratigraphic markers might be the more important location. Measurements from core are problematic in lots of ways:  borehole is not a vertical straight line, the sediment extracted in the core expands when brought to the surface; incomplete recovery...  
Truth in advertising is the key to good metadata. Thus there are concepts like 'true vertical depth relative to EPSG:5100', or 'depth below KB along borehole trace', '500 millibar elevation', 'depth below base of Sasquetin Formation measured in recovered core'.  
Interoperability for location-based discovery requires consistency, and given the challenges this comes as the cost of some loss of accuracy translating the available vertical location information into a global elevation reference system (e.g. [EPSG:5715](https://epsg.io/5715).
